### PR TITLE
Update Taylor

### PR DESCRIPTION
**Pull request by Jiaqi-beep**
I have edited the TaylorDiagram constructor and added 11 instance methods to the class to recreate taylor_2, taylor_3, and taylor_6_3. All changes are in ref_taylor.py and recreated plots can be called on line 566. I'd love to take some advice on things like comments, doc strings conventions, or edits to instance method arguments structure to make them more user-friendly.

- [taylor_2](https://www.ncl.ucar.edu/Applications/Images/taylor_2_lg.png)
image
- [taylor_3](https://www.ncl.ucar.edu/Applications/Images/taylor_3_lg.png)
image
- [taylor_6_3](https://www.ncl.ucar.edu/Applications/Images/taylor_6_3_lg.png)
image

### CODE REVIEW

File: src/taylor/taylor.py [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+        ------
+        None
+        """
+        self._ax.axis['top', 'right',
+                      'left'].major_ticklabels.set_fontsize(ticklabel_fontsize)
+        self._ax.axis['top', 'right'].label.set_fontsize(axislabel_fontsize)
+        self._ax.axis['top', 'right'].label.set_pad(axislabel_pad)
+
+
+###############################################################################
```
Comment (michaelavs) [inline src/taylor/taylor.py]: Will everything from here down be split into something for geocat-examples?

File: src/taylor/taylor.py [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+
+    # Add x axis grid
+    dia.add_xgrid(np.array([0.6, 0.9]))
+
+    # Add figure title
+    fig.suptitle("Example", size=24)
+
+    return fig
+
+    # p dataset
```
Comment (michaelavs) [inline src/taylor/taylor.py]: I think from line 591 to 616 can be deleted

**Review by michaelavs — COMMENTED** [review]
This is looking great! Just some formatting comments but nothing functional

File: src/geocat/viz/taylor/taylor.py [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+            Optional reference label string indentifier
+
+        stdRange : Tuple, optional
+            Optional stddev axis extent
+
+        stdLevel : list, optional
+            Optional list of tick locations for stddev axis
+        """
+
+        from matplotlib.projections import PolarAxes
```
Comment (michaelavs) [inline src/geocat/viz/taylor/taylor.py:53]: I think these imports should all be in one place at the top with the other imports

File: src/geocat/viz/taylor/taylor.py [hunk 4]
```diff
@@ -0,0 +1,600 @@
...
+        self.modelMarkerSet = []
+        
+        # Set number for models outside axes
+        self.modelOutside = -1
+        
+    def biasToMarkerSize(self, biasarr):
+        '''Helper function to return marker size and sign based on input bias'''
+        out_arr = []
+        for bias in biasarr:
+            ab=abs(bias)
```
Comment (michaelavs) [inline src/geocat/viz/taylor/taylor.py:160]:
```suggestion
            ab = abs(bias)
```
Just for formatting

File: src/geocat/viz/taylor/taylor.py [hunk 5]
```diff
@@ -0,0 +1,600 @@
...
+        self.referenceLine = h
+
+        # Collect sample points for latter use (e.g. legend)
+        self.modelMarkerSet = []
+        
+        # Set number for models outside axes
+        self.modelOutside = -1
+        
+    def biasToMarkerSize(self, biasarr):
+        '''Helper function to return marker size and sign based on input bias'''
```
Comment (michaelavs) [inline src/geocat/viz/taylor/taylor.py:157]:
```suggestion
        """Helper function to return marker size and sign based on input bias"""
```
For consistency of docstring formatting

File: src/geocat/viz/taylor/taylor.py [hunk 6]
```diff
@@ -0,0 +1,600 @@
...
+        edgecolors='blue',
+        facecolors='none',
+        label='Data B')
+    
+    # Change properties of model labels to add bounding boxes
+    for txt in modelTextsA:
+        txt.set_bbox(dict(facecolor='red', edgecolor='none',
+                          pad=0.05, boxstyle='square'))
+        txt.set_color('white')
+    for txt in modelTextsB:
```
Comment (michaelavs) [inline src/geocat/viz/taylor/taylor.py:586]: I'm not sure if there is a way to edit the edge, face, and text color for case B data, but I do notice a difference between point 5 on NCL and this version. If there isn't a way to work around the colors being set for every point on the diagram, don't worry about it

**Review by michaelavs — COMMENTED** [review]
Looking great overall!

File: src/geocat/viz/taylor/taylor.py (hunk 6 above)
Comment (Jiaqi-beep) [reply src/geocat/viz/taylor/taylor.py:690]: We could potentially, but I personally don't understand why the text format for the labels in the NCL is different. It seemed unnecessary to me.

**Comment by Jiaqi-beep** [comment]
[taylor_8](https://www.ncl.ucar.edu/Applications/Images/taylor_8_lg.png)
image

**Review by anissa111 — APPROVED** [review]
This looks done from my end! Thank you for all your hard work on this @Jiaqi-beep!

**Pull request merged**